src/UCTNode.cpp

```diff
@@ -334,8 +343,13 @@ UCTNode* UCTNode::uct_select_child(int color) {
             continue;
         }
 
-        // get_eval() will automatically set first-play-urgency
-        auto winrate = child->get_eval(color);
+        auto winrate = 0.5f;
+        if (child->get_visits() > 0) {
+            winrate = child->get_eval(color);
+        } else { // First play urgency
+            winrate = child->get_eval(color) - cfg_fpu_reduction*std::sqrt(total_visited_policy);
```

Comment:
This can be cached like numerator 

auto fpu_policy_reduction = cfg_fpu_reduction * std::sqrt(total_visited_policy);